### PR TITLE
Feature/restaurants

### PR DESCRIPTION
**Pull request by emilio-gambino**
Adds restaurants and their reviews to our app

### CODE REVIEW

File: app/src/main/java/com/github/sdp/ratemyepfl/database/ItemsRepository.kt [hunk 1]
```diff
@@ -4,33 +4,40 @@ import com.github.sdp.ratemyepfl.model.items.Classroom
...
     companion object {
         private const val TAG_COURSES = "CourseRepository"
         private const val TAG_ROOMS = "ClassroomRepository"
+        private const val TAG_RESTAURANTS= "RestaurantRepository"
     }
 
-    override suspend fun getCourses(): List<Course?> {
-        return getLimit(collectionCourses, 50)
-            .mapNotNull { obj -> obj.toCourse() }
+    override suspend fun getCourses() : List<Course?> {
```
Comment (mbouilloud) [inline app/src/main/java/com/github/sdp/ratemyepfl/database/ItemsRepository.kt:23]: Why did you change this part, I think Julien refactored some part to avoid duplication

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/RestaurantListActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+@HiltAndroidTest
+class RestaurantListActivityTest {
+
+    @get:Rule(order = 0)
+    val hiltRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val testRule = ActivityScenarioRule(RestaurantListActivity::class.java)
+
+//    @Test
```
Comment (mbouilloud) [inline app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/RestaurantListActivityTest.kt:31]: You can correct this test now, before merging.

File: app/src/main/java/com/github/sdp/ratemyepfl/database/ItemsRepository.kt (hunk 1 above)
Comment (emilio-gambino) [reply app/src/main/java/com/github/sdp/ratemyepfl/database/ItemsRepository.kt:23]: Merging error, my bad :(

**Review by mbouilloud — APPROVED** [review]

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/RestaurantListActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    @get:Rule(order = 0)
+    val hiltRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val testRule = ActivityScenarioRule(RestaurantListActivity::class.java)
+
+    @Test
+    fun isRecyclerViewVisibleOnActivityLaunch() {
+        Espresso.onView(ViewMatchers.withId(R.id.reviewableRecyclerView))
```
Comment (jmettler27) [inline app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/RestaurantListActivityTest.kt:38]: You could rewrite these tests as `onView(withId(...))` for instance to make them more comfortable to read, and apply that to the other tests as well

File: app/src/test/java/com/github/sdp/ratemyepfl/model/items/RestaurantTest.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package com.github.sdp.ratemyepfl.model.items
+
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class RestaurantTest {
```
Comment (jmettler27) [inline app/src/test/java/com/github/sdp/ratemyepfl/model/items/RestaurantTest.kt:9]: Great tests!

**Review by jmettler27 — APPROVED** [review]

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/RestaurantListActivityTest.kt (hunk 3 above)
Comment (emilio-gambino) [reply app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/RestaurantListActivityTest.kt:38]: Yes, my bad, Mathias will change it in his refactoring !
